src/main/java/com/azure/cosmos/examples/changefeed/SampleChangeFeedEstimator.java

```diff
@@ -0,0 +1,172 @@
+    private static final ObjectMapper OBJECT_MAPPER = Utils.getSimpleObjectMapper();
+    protected static Logger logger = LoggerFactory.getLogger(SampleChangeFeedEstimator.class);
+
+    public static void main(String[] args) {
+        logger.info("Begin Sample");
+        try {
+            //Summary of the next four commands:
+            //-Create an asynchronous Azure Cosmos DB client and database so that we can issue async requests to the DB
+            //-Create a "feed container" and a "lease container" in the DB
+            logger.info("Create CosmosClient");
+            CosmosAsyncClient client = getCosmosClient();
+
+            logger.info("Create sample's database: " + DATABASE_NAME);
+            CosmosAsyncDatabase cosmosDatabase = createNewDatabase(client, DATABASE_NAME);
+
+            logger.info("Create container for documents: " + COLLECTION_NAME);
+            CosmosAsyncContainer feedContainer = createNewCollection(client, DATABASE_NAME, COLLECTION_NAME);
+
+            logger.info("Create container for lease: " + COLLECTION_NAME + "-leases");
+            CosmosAsyncContainer leaseContainer = createNewLeaseCollection(client, DATABASE_NAME, COLLECTION_NAME + "-leases");
+
+            //Model of a worker thread or application which leases access to monitor one or more feed container
+            //partitions via the Change Feed. In a real-world application you might deploy this code in an Azure function.
+            //The next line causes the worker to create and start an instance of the Change Feed Processor. See the implementation of getChangeFeedProcessor() for guidance
+            //on creating a handler for Change Feed events. In this stream, we also trigger the insertion of 10 documents on a separate
+            //thread.
+            logger.info("Start Change Feed Processor on worker (handles changes asynchronously)");
+            ChangeFeedProcessor changeFeedProcessorInstance = new ChangeFeedProcessorBuilder()
+                .hostName("SampleHost_1")
+                .feedContainer(feedContainer)
+                .leaseContainer(leaseContainer)
+                .handleChanges(handleChangesWithLag())
+                .buildChangeFeedProcessor();
+
+            changeFeedProcessorInstance.start()
+                                       .subscribeOn(Schedulers.boundedElastic())
+                                       .subscribe();
+
+            //These two lines model an application which is inserting ten documents into the feed container
+            logger.info("Start application that inserts documents into feed container");
+            createNewDocumentsCustomPOJO(feedContainer, 10, Duration.ofSeconds(3));
+
+            Thread.sleep(Duration.ofSeconds(4).toMillis());
+
+            AtomicInteger totalLag = new AtomicInteger();
+
+            // <EstimatedLag>
+            Mono<Map<String, Integer>> estimatedLagResult = changeFeedProcessorInstance
```

Comment:
The current implementation of this sample does not capture the real intend of the getCurrentState() (btw, getEstimatedLag() call below should be changed as that API is being deprecated).
We need basically two separate sections for this sample to work, ideally creating and running two application threads as if the customer has two distinct instances of their application, one that does the change feed processing, the other that does the current state monitoring. https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/ChangeFeedProcessorTest.java#L365 captures some of that but not in separate threads.